### PR TITLE
e2e-builds: Private package for build logic of e2e

### LINKED ISSUE #32654: Standardize `build` Scripts
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

In order to ensure consistency within the monorepo build tooling, all of the `build` tasks should be examined. Each of them should perform anything necessary to build the project, while also being isolated from any other project in the monorepo. For the plugins, we should make sure to set the `build_step` configuration for Woorelease, as well as prevent `build` from outputting a ZIP file.

- [x] packages/js/admin-e2e-tests
- [x] packages/js/api
- [x] packages/js/api-core-tests
- [x] packages/js/components
- [x] packages/js/csv-export
- [x] packages/js/currency
- [x] packages/js/customer-effort-score
- [x] packages/js/data
- [x] packages/js/date
- [x] packages/js/dependency-extraction-webpack-plugin
- [x] packages/js/e2e-core-tests https://github.com/woocommerce/woocommerce/pull/32680
- [x] packages/js/e2e-environment https://github.com/woocommerce/woocommerce/pull/32680
- [x] packages/js/e2e-utils https://github.com/woocommerce/woocommerce/pull/32680
- [x] packages/js/eslint-plugins
- [x] packages/js/experimental
- [x] packages/js/explat
- [x] packages/js/js-tests
- [x] packages/js/navigation
- [x] packages/js/notices
- [x] packages/js/number
- [x] packages/js/onboarding
- [x] packages/js/style-build
- [x] packages/js/tracks
- [x] plugins/woocommere https://github.com/woocommerce/woocommerce/pull/32689
- [x] plugins/woocommerce-admin
- [x] plugins/woocommerce-beta-tester https://github.com/woocommerce/woocommerce/pull/32711

* If anything is added between the creation and completion of this issue, make sure to check that too!

### PR DESCRIPTION
**Pull request by psealock**
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

e2e packages used logic found in `packages/js/bin` to create builds. In an effort to isolate packages before converting the monorepo to Turborepo, that logic should be properly brought in as a dependency rather than referencing a bin folder that doesn't really belong in `packages/js`.

This PR creates a private package and declares a proper dependency in the consuming packages' `package.json`.

Partially fixes https://github.com/woocommerce/woocommerce/issues/32654

### How to test the changes in this Pull Request:

1. `pnpm install`
2. View the logs and confirm that 3 `e2e-*` packages are built without error.

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [x] Have you written new tests for your changes, as applicable?
* [x] Have you successfully run tests with your changes locally?
* [x] Have you created a changelog file by running `pnpm nx affected --target=changelog`?

### FOR PR REVIEWER ONLY:

* [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: packages/js/e2e-builds/.npmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+package-lock=false
```
Comment (roykho) [inline packages/js/e2e-builds/.npmrc]: Needs a new line at the end of the file.

File: packages/js/e2e-builds/project.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+     "targets": {
+         "changelog": {
+             "executor": "./tools/executors/changelogger:changelog",
+             "options": {
+                 "action": "add",
+                 "cwd": "packages/js/e2e-builds"
+             }
+         }
+     }
+ }
```
Comment (roykho) [inline packages/js/e2e-builds/project.json]: Needs a new line at the end of the file.

**Review by roykho — CHANGES_REQUESTED** [review]
Tested and working well. No errors building the packages. Just left small nitpick comments.

**Comment by psealock** [comment]
Thanks @roykho, I resolved the whitespace issues

**Review by roykho — APPROVED** [review]
LGTM! Nice work.

**Pull request merged**